components/EventDetail.js

```diff
@@ -9,43 +14,173 @@ const StyledContainer = styled.div`
   max-width: 36rem;
   margin-inline: auto;
 `;
+
 const StyledHeaderImage = styled(Image)`
   width: 100%;
 `;
+
 const StyledTitle = styled.p``;
 const StyledEventInfoContainer = styled.div`
   display: grid;
   grid-template-columns: 1fr 1fr;
   justify-content: space-between;
   align-items: baseline;
 `;
+
 const StyledEventInfo = styled.p`
   justify-self: ${(props) => (props.$right ? "end" : "start")};
 `;
+
 const StyledDescription = styled.p``;
 
-export default function EventDetail({ event }) {
-  const { day, month, year, time } = getDate(event.startDateTime);
+const initialState = {
+  title: "",
+  city: "",
+  category: "",
+  startDateTime: "",
+  organizer: "",
+  description: "",
+};
+
+export default function EventDetail() {
+  const router = useRouter();
+  const { id } = router.query;
+  const { mutate, data: event } = useSWR(`/api/events/${id}`);
+
+  // destructuring a formatted date of the event object
+  const { day, month, year, time } = getDate(event?.startDateTime);
 
+  const [formData, setFormData] = useState(initialState);
+
+  // toggle for Edit Mode
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  // set input fields to the current event data
+  useEffect(() => {
+    setFormData({
+      title: event.title,
+      city: event.city,
+      category: event.category,
+      startDateTime: event.startDateTime,
+      organizer: event.organizer,
+      description: event.description,
+    });
+  }, [event]);
+
+  // handle functions
+  async function handleSubmit(jsEvent) {
+    jsEvent.preventDefault();
+    await updateEvent(id, formData);
+    setIsEditMode(false);
+    mutate();
+  }
+  function handleChange(jsEvent, target) {
+    setFormData((prevFormData) => {
+      return {
+        ...prevFormData,
+        [target]: jsEvent.target.value,
+      };
+    });
+  }
   return (
-    <StyledContainer>
-      <StyledHeaderImage
-        src={event.imageUrl}
-        alt={event.title}
-        width={260}
-        height={260}
-      />
-
-      <StyledEventInfoContainer>
-        <StyledTitle>{event.title}</StyledTitle>
-        <StyledEventInfo $right>{event.category}</StyledEventInfo>
-        <StyledEventInfo>{event.city}</StyledEventInfo>
-        <StyledEventInfo
-          $right
-        >{`${day}. ${month} ${year}, ${time} Uhr`}</StyledEventInfo>
-        <StyledEventInfo>{event.organizer}</StyledEventInfo>
-      </StyledEventInfoContainer>
-      <StyledDescription>{event.description}</StyledDescription>
-    </StyledContainer>
+    <>
+      <StyledContainer>
```

Comment:
I suggest to change the structure of the following code. There are a few things, you should consider:

When the user clicks on the edit-button, a form should appear with prefilled data. Unfortunately, you are not using a form here 🙁 You have already a state indicating if the user is in edit-mode. You can simply display the details of the event as they are and only if `isEditMode` is true, you can display a form with the prefilled data. The form you can submit as usual, when the user clicks on "save". In this way, you will have a proper HTML structure (a form with labels, inputs and submit button) and your code will have a better structure and becomes more readable and probably a bit shorter as well.

Additionally, as you will implement a form, you won't need a state to save your form data (also the `useEffect` and `handleChange()` function won't be necessary anymore). If the user is in edit-mode, you can fill all input fields with the already saved data via the `value` attribute. Once the user submits the form, you can get the data of all form fields (including the changes, which have been made) and save them to your database.

Reply:
Thank you, yes thats much easier. see our solution in the next version.